homeassistant/components/tplink_omada/binary_sensor.py

```diff
@@ -107,6 +128,11 @@ def __init__(
 
         self._attr_name = f"Port {config.port_number} {config.name_suffix}"
 
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._handle_coordinator_update()
```

Comment:
We shouldn't update state twice. Please extract the attribute updates to a separate method from `_handle_coordinator_update` that we can call here.

Reply:
Done. I think I've got this right now - it's not super clear in my mind when `async_write_ha_state` needs to be called.